### ISSUE TITLE
Error when trying to retrieve a Context Source Registration Subscription

### ISSUE THREAD

**Issue by HoucemKacem** [ScorpioBroker/ScorpioBroker#181]
Hello,

I started Scorpio (FIWARE_7.9.1 release) with docker-compose. 

I created a Context Source Registration Subscription with the following payload:
```
{
   "id":"urn:ngsi-ld:Subscription:cSourceRegistration:01",
   "type":"Subscription",
   "entities":[
      {
         "type":"Beekeeper"
      }
   ],
   "notification":{
      "format":"keyValues",
      "endpoint":{
         "uri":"http://my.endpoint.org/notify",
         "accept":"application/json"
      }
   },
   "@context":[
      "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.3.jsonld"
   ]
}
```

It was successfully created since I received a 201 response.

Then when I tried to retrieve it with a GET request on `/ngsi-ld/v1/csourceSubscriptions/urn:ngsi-ld:Subscription:cSourceRegistration:01`, I received the following response:

```
{
    "timestamp": "2021-01-06T15:00:55.485+0000",
    "status": 500,
    "error": "Internal Server Error",
    "message": "No message available",
    "path": "/ngsi-ld/v1/csourceSubscriptions/urn:ngsi-ld:Subscription:cSourceRegistration:01"
}
```

The logs shows:
```
regmgr_1     | 2021-01-06T15:02:49,347 ERROR [http-nio-1030-exec-2] o.a.j.l.DirectJDKLog: Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.NullPointerException] with root cause
regmgr_1     | java.lang.NullPointerException
regmgr_1     | 	at eu.neclab.ngsildbroker.commons.tools.HttpUtils.generateReply(HttpUtils.java:943)
regmgr_1     | 	at eu.neclab.ngsildbroker.commons.tools.HttpUtils.generateReply(HttpUtils.java:859)
regmgr_1     | 	at eu.neclab.ngsildbroker.commons.tools.HttpUtils.generateReply(HttpUtils.java:854)
regmgr_1     | 	at eu.neclab.ngsildbroker.commons.tools.HttpUtils.generateReply(HttpUtils.java:848)
regmgr_1     | 	at eu.neclab.ngsildbroker.registryhandler.controller.RegistrySubscriptionController.getSubscriptions(RegistrySubscriptionController.java:127)
regmgr_1     | 	at sun.reflect.GeneratedMethodAccessor149.invoke(Unknown Source)
regmgr_1     | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
regmgr_1     | 	at java.lang.reflect.Method.invoke(Method.java:498)
regmgr_1     | 	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
regmgr_1     | 	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
regmgr_1     | 	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
regmgr_1     | 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:893)
regmgr_1     | 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:798)
regmgr_1     | 	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
regmgr_1     | 	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
regmgr_1     | 	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
regmgr_1     | 	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
regmgr_1     | 	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
regmgr_1     | 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:634)
regmgr_1     | 	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
regmgr_1     | 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.boot.actuate.web.trace.servlet.HttpTraceFilter.doFilterInternal(HttpTraceFilter.java:88)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
regmgr_1     | 	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)
regmgr_1     | 	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationProcessingFilter.doFilter(OAuth2AuthenticationProcessingFilter.java:176)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:74)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
regmgr_1     | 	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
regmgr_1     | 	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
regmgr_1     | 	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:94)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.filterAndRecordMetrics(WebMvcMetricsFilter.java:114)
regmgr_1     | 	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:104)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
regmgr_1     | 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
regmgr_1     | 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
regmgr_1     | 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
regmgr_1     | 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
regmgr_1     | 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
regmgr_1     | 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
regmgr_1     | 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
regmgr_1     | 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
regmgr_1     | 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
regmgr_1     | 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
regmgr_1     | 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
regmgr_1     | 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
regmgr_1     | 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
regmgr_1     | 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
regmgr_1     | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
regmgr_1     | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
regmgr_1     | 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
regmgr_1     | 	at java.lang.Thread.run(Thread.java:748)
```
Is there a problem in my request ? Thanks.

**Comment by amitrnec** [comment]
Fixed this issue in #183

**Comment by amitrnec** [comment]
merged into development branch.
